### PR TITLE
avoid passing incompatible pointer types to message_get_bytes_header

### PR DESCRIPTION
**Pull request by scareything**
fixes builds on gcc 14.

### CODE REVIEW

**Review by dovholuknf — APPROVED** [review]

File: inc_internal/message.h [hunk 1]
```diff
@@ -90,7 +90,7 @@ bool message_get_int32_header(message *m, int header_id, int32_t *v);
 
 bool message_get_uint64_header(message *m, int header_id, uint64_t *v);
 
-bool message_get_bytes_header(message *m, int header_id, const char **ptr, size_t *len);
+bool message_get_bytes_header(message *m, int header_id, uint8_t **ptr, size_t *len);
```
Comment (ekoby) [inline inc_internal/message.h:93]: that is a bad idea -- this returns the pointer to data owned by the message -- therefore marked const to the caller.
The caller cannot assume that the data is modifiable.

File: library/bind.c [hunk 2]
```diff
@@ -381,7 +381,7 @@ static void process_dial(struct binding_s *b, message *msg) {
     bool caller_id_sent = message_get_bytes_header(msg, CallerIdHeader, &source_identity, &source_identity_sz);
 
     ziti_client_ctx clt_ctx = {0};
-    message_get_bytes_header(msg, AppDataHeader, (uint8_t **) &clt_ctx.app_data, &clt_ctx.app_data_sz);
+    message_get_bytes_header(msg, AppDataHeader, &clt_ctx.app_data, &clt_ctx.app_data_sz);
```
Comment (ekoby) [inline library/bind.c:384]: it would be safer to make a copy here -- otherwise app_data points to garbage once msg is discarded

**Review by ekoby — CHANGES_REQUESTED** [review]
current `app_data` header usage seems unsafe

File: library/bind.c (hunk 2 above)
Comment (scareything) [reply library/bind.c:384]: Just to be clear, the variable scoping has not been changed in this PR. 

I believe `cit_ctx->app_data` will be assigned the pointer from the message header. `clt_ctx` is a stack variable, which is only passed to the client callback. We do [document that the client context is only valid while the callback is on the stack](https://github.com/openziti/ziti-sdk-c/blob/main/includes/ziti/ziti.h#L337).

File: inc_internal/message.h (hunk 1 above)
Comment (scareything) [reply inc_internal/message.h]: changed signature back to const.

File: library/bind.c (hunk 2 above)
Comment (ekoby) [reply library/bind.c]: I think to make it more clear both `clt_ctx` and `ctl_ctx.app_data` should be `const` in the callback

**Review by ekoby — APPROVED** [review]

**Pull request merged**